crates/starlight/src/vm/context.rs

```diff
@@ -144,19 +144,17 @@ impl Context {
     }
 }
 impl GcPointer<Context> {
-    pub fn register_native_reference(reference: usize) {
-        unsafe {
-            VM_NATIVE_REFERENCES.push(reference);
-        }
+    pub fn register_external_reference(&mut self, reference: usize) {
```

Comment:
We should get rid of this function. As I said references should be registered only when VM is created.